tools/c7n_azure/c7n_azure/query.py

```diff
@@ -43,29 +50,66 @@ def __init__(self, session_factory):
 
     def filter(self, resource_manager, **params):
         m = resource_manager.resource_type
-        enum_op, list_op, extra_args = m.enum_spec
 
-        if extra_args:
-            params.update(extra_args)
+        # Generic Handling
+        if hasattr(m, 'resource_type'):
+            session = resource_manager.get_session()
+            subscriptionId = session.get_subscription_id()
+            url = f'/subscriptions/{subscriptionId}/providers/{m.resource_type}'
+            type_parts = m.resource_type.split('/')
 
-        params.update(m.extra_args(resource_manager))
+            config = AzureConfiguration(base_url='https://management.azure.com/')
+            config.credentials = session.credentials
+            client = ServiceClient(config.credentials, config)
 
-        try:
-            op = getattr(getattr(resource_manager.get_client(), enum_op), list_op)
-            result = op(**params)
+            # Construct parameters
+            query_parameters = {
+                'api-version': session.resource_api_version(
```

Comment:
I feel like this is worth a discussion, re whether we should just be using a known named profile vs interactive querying. it came up in the sdk/strip pr, its actually nicer here since with this change we can lesson our burden on the sdk bits so perhaps this isn't as bad, but it still feels like we're playing roulette on apis instead of using known good set.

Reply:
re known named profiles .. https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-common/azure/profiles/__init__.py

Reply:
Yeah I'm afraid this was their attempt to stop duplication of clients just to change the version but that only a tiny number of clients are using it so far.  I think clients are opting into using this method on a case by case basis and they own managing their version numbers.

Reply:
Oh I get it - the only profiles they are maintaining are non-public-cloud profiles - everything else just defaults to `latest` which means `whatever the latest version number published in your SDK is` (which requires all the SDK's, and isn't particularly meaningful).

Reply:
What if we just always grab latest, but provide an override field where users can provide the API-Version they want to target. That way we don't need to bake it into our resources.

Just throwing out an another idea, we could have a config file that pins down specifics API Versions that is generated by our CI, where all of our E2E validation tests have passed.

Reply:
Yes we could commit our own profiles, although I'm not sure if there is much precedent for it, if you install Azure CLI or the SDK today you always get the latest API version too.  Main difference is you have more control over when you update those.  

Until a month ago (Poetry) every time you installed custodian you got the latest also.

Would be easy for us to do though, and the maintenance is pretty low - these versions are supported for a long time.

Reply:
Yeah after thinking through it more I'm increasingly convinced best implementation would be for us to simply persist the JSON result from this Version API in the source tree and rev it as needed periodically.    The only code that changes is the version lookup method, which could still maintain the old code as a fallback if it is useful to do so.

I'll open another issue recapping some of the discussion as part of finishing this PR.

Reply:
wouldn't that also need access to all the azure clouds endpoints to map out our own endpoints? if thats viable sounds good. 

to clarify wrt to azure installs, c7n_azure packages published to pypi have been frozen for a while (at least 6 months), poetry is really about giving us better tooling to do that across all the pypi package artifacts as well as better dependency management across the tree. ie. install c7n_azure today is a repeatable process with the last few releases at least with respect to the azure sdks.

all that said, ditching the sdks by and large is a huge win on many fronts, so +1.. I'm glad we can land here.